homeassistant/components/netgear/router.py

```diff
@@ -0,0 +1,283 @@
+"""Represent the Netgear router and its devices."""
+from datetime import timedelta
+import logging
+
+from pynetgear import Netgear
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC, format_mac
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect,
+    async_dispatcher_send,
+)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.util import dt as dt_util
+
+from .const import CONF_CONSIDER_HOME, DEFAULT_CONSIDER_HOME, DOMAIN, MODELS_V2
+from .errors import CannotLoginException
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_api(
+    password: str,
+    host: str = None,
+    username: str = None,
+    port: int = None,
+    ssl: bool = False,
+) -> Netgear:
+    """Get the Netgear API and login to it."""
+    api: Netgear = Netgear(password, host, username, port, ssl)
+
+    if not api.login():
+        raise CannotLoginException
+
+    return api
+
+
+async def async_setup_netgear_entry(
+    hass: HomeAssistantType,
+    entry: ConfigEntry,
+    async_add_entities,
+    entity_class_generator,
```

Comment:
Missing typing

Reply:
Both are functions, I have very little experiance with typing and don't know how to do typing for functions.
Could you help me out?

Reply:
For async_add_entities we have a type for that built-in

`from homeassistant.helpers.entity_platform import AddEntitiesCallback`

Reply:
For `entity_class_generator` its going to be a `Callable` that takes `NetgearRouter, dict`

https://docs.python.org/3/library/typing.html#typing.Callable

Reply:
Thanks for the help, appreciate it!